### DISCUSSION TITLE
unsupported op . for object type

### DISCUSSION THREAD

**Discussion by sam-heilbron** [cuelang/cue#918] [Q&A]
I'm trying to better understand what could cause this error when converting protos to open api schemas. I have the following message:

```proto
message MockMessage {
    NestedOneOf option = 1;
}

message NestedOneOf {
    oneof option {
        string option_a = 1;
        string option_b = 2;
    }

    string field = 3;

    repeated string multiple_a = 4;
    repeated string multiple_b = 5;
    repeated string multiple_c = 6;

    bool choice = 7;
}
```

When I include this message in some projects it works, but in others it produces the `unsupported op . for object type` error. Does anyone have tips about what that error means?

**Comment by mpvl** [comment]
Can you give the steps to reproduce the error you see?

**Comment by myitcv** [comment]
@sam-heilbron please can you give this a try with `v0.4.0`? There were some fixes that went into that release related to `oneOf`, detected as part of the Istio project.

Incidentally, if you rely on OpenAPI support I would encourage you to add your project to [`unity`](https://github.com/cue-sh/unity) corpus. That will give us increased test coverage when it comes to changes in that space (a fairly major rewrite is going to be required at some point, along with a new API).

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/918.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
